pkg/api/pod/util.go

```diff
@@ -440,6 +440,14 @@ func useRelaxedEnvironmentVariableValidation(podSpec, oldPodSpec *api.PodSpec) b
 		podEnvVarNames = gatherPodEnvVarNames(podSpec)
 	}
 
+	// If the changed field is not the environment variable name and
+	// the RelaxedEnvironmentVariableValidation feature gate is not enabled,
+	// we will still use relaxed validation to prevent workload
+	// failures when updating other fields after cluster rollback.
+	if oldPodEnvVarNames.Equal(podEnvVarNames) {
```

Comment:
can we add a test for this behavior please?

Reply:
Ah, I realized that I had already implemented it when working on the alpha version.
Ref: https://github.com/kubernetes/kubernetes/blob/e673417529d9d2d2dd04f28f8437cbe1c403dfbe/pkg/api/pod/util.go#L511-L532

So this code is not needed, and I have already revert.